exp/controllers/gcpmanagedcluster_controller.go

```diff
@@ -212,15 +210,15 @@ func (r *GCPManagedClusterReconciler) reconcile(ctx context.Context, clusterScop
 	record.Event(clusterScope.GCPManagedCluster, "GCPManagedClusterReconcile", "Ready")
 
 	controlPlaneEndpoint := clusterScope.GCPManagedControlPlane.Spec.Endpoint
+	clusterScope.SetControlPlaneEndpoint(controlPlaneEndpoint)
+
 	if controlPlaneEndpoint.IsZero() {
 		log.Info("GCPManagedControlplane does not have endpoint yet. Reconciling")
 		record.Event(clusterScope.GCPManagedCluster, "GCPManagedClusterReconcile", "Waiting for control-plane endpoint")
-		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
do we still want to requeue in this case?

Reply:
No we don't as there is nothing left for the controller to do until the control plane has updated its endpoint....and we have a watch for that.